test/resources/auction_examples.js

```diff
@@ -0,0 +1,335 @@
+    owners: [0, 1],
+    tokenIdsForPrice: [0, 1],
+    // buyVolumes: [toETH(1), new BN("184000000000000000815")],
+    buyVolumes: [toETH(1), new BN("184000000000000000000")],
+    sellVolumes: [new BN("184368552736921106106"), toETH(1)],
+  }
+]
+
+// /////--------------- Two orders A -> B and two orders B -> A 
+
+// const fiveHunnit = toETH(500)
+// const tenThouzy = toETH(10000)
+
+// const doubleDoubleTrade = {
+//   deposits: [
+//     { amount: fiveHunnit, token: 0, user: 0 },
+//     { amount: fiveHunnit, token: 1, user: 1 },
+//   ],
+//   orders: [
+//     { sellToken: 0, buyToken: 1, sellAmount: fiveHunnit, buyAmount: tenThouzy, user: 0 },
+//     { sellToken: 1, buyToken: 0, sellAmount: fiveHunnit, buyAmount: tenThouzy, user: 1 },
+//     { sellToken: 0, buyToken: 1, sellAmount: fiveHunnit, buyAmount: tenThouzy, user: 1 },
+//     { sellToken: 1, buyToken: 0, sellAmount: fiveHunnit, buyAmount: tenThouzy, user: 2 }
+//   ]
+// }
+
+// const doubleDoubleTradeSolutions = [
+//   {
+//     name: "Not sure about this one...",
+//     prices: [1, 1].map(toETH),
+//     owners: [0, 1, 0, 1],
+//     buyVolumes: [100, 99, 98, 97].map(toETH),
+//     tokenIdsForPrice: [0, 1],
+//   },
+// ]
+
+// /////--------------- Basic Ring Trade example A -> B -> C
+
+// buy amounts: [ '990000000000000000', '990000000000000000', '990000000000000000' ]
+// sell amounts: [ '1000000000000000000', '1000000000000000000', '1000000000000000000' ]
+// prices: [ '1000000000000000000', '1000000000000000000', '1000000000000000000' ]
+// buy volumes: [ '999000000000000000', '998001000000000000', '997002999000000000' ]
+// sell volumes: [ '1000000000000000000', '999000000000000000', '998001000000000000' ]
+// token conservation: [ '2997001000000000', '0', '0' ]
+// order utilities: [
+//   '9000000000000000000000000000000000',
+//   '8991000000000000000000000000000000',
+//   '8982009000000000000000000000000000'
+// ]
+// order disregarded utilities: [
+//   '0',
+//   '9009009009009009009009009009009',
+//   '18009009009009009009009009009009'
+// ]
+
+const sell = new BN("1000000000000000000")
+const buy = new BN("990000000000000000")
+
+const basicRingTrade = {
+  deposits: [
+    { amount: sell, token: 0, user: 0 },
+    { amount: sell, token: 1, user: 1 },
+    { amount: sell, token: 2, user: 2 },
+  ],
+  orders: [
+    { sellToken: 0, buyToken: 1, sellAmount: sell, buyAmount: buy, user: 0 },
+    { sellToken: 1, buyToken: 2, sellAmount: sell, buyAmount: buy, user: 1 },
+    { sellToken: 2, buyToken: 0, sellAmount: sell, buyAmount: buy, user: 2 }
+  ]
+}
+
+const basicRingTradeSolutions = [
+  {
+    name: "Simple Ring",
+    prices: [1, 1, 1].map(toETH),
+    owners: [0, 1, 2],
+    buyVolumes: [new BN("999000000000000000"), new BN("998001000000000000"), new BN("997002999000000000")],
+    sellVolumes: [new BN("1000000000000000000"), new BN("999000000000000000"), new BN("998001000000000000")],
+    tokenIdsForPrice: [0, 1, 2],
```

Comment:
I would use `toETH(1)`, `feeSubtracted(toETH(1))` for `99900..0`, `feeSubtractedN(toETH(1), 2)` for `9980010..0` and `feeSubtractedN(toETH(1), 3)` for `9970029990..0` where `feeSubtractedN` subtracts the fee *n* times.

Reply:
Also, we calculate sell volumes so we don't need to include them in the object here

Reply:
Your comments are always appearing like two lines below the suggestion and github can't recognise when the suggestions have been included.